src/main/java/nextstep/session/domain/EnrollmentPolicy.java

```diff
@@ -0,0 +1,55 @@
+package nextstep.session.domain;
+
+import nextstep.session.InvalidEnrollmentPolicyException;
+
+public class EnrollmentPolicy {
+
+    private final static int MIN_ENROLLMENT = 0;
+    private final static int MIN_FEE = 0;
+    private final PriceType priceType;
+    private final int maxEnrollment;
+    private final int fee;
+
+
+    private EnrollmentPolicy(PriceType priceType, int maxEnrollment, int fee)
+        throws InvalidEnrollmentPolicyException {
+        if (!validate(maxEnrollment, fee)) {
+            throw new InvalidEnrollmentPolicyException(
+                String.format("최소수강인원은 %d명 이상, 최소 수강료는 %d원 이상이여야 합니다.", MIN_ENROLLMENT, MIN_FEE));
+        }
+        this.priceType = priceType;
+        this.maxEnrollment = maxEnrollment;
+        this.fee = fee;
+    }
+
+    public static EnrollmentPolicy createFreePolicy() throws InvalidEnrollmentPolicyException {
+        return new EnrollmentPolicy(PriceType.FREE, MIN_ENROLLMENT, 0);
+    }
+
+    public static EnrollmentPolicy createPaidPolicy(int maxEnrollment, int fee)
+        throws InvalidEnrollmentPolicyException {
+        return new EnrollmentPolicy(PriceType.PAID, maxEnrollment, fee);
+    }
+
+    private boolean validate(int maxEnrollment, int fee) {
+        if (maxEnrollment < MIN_ENROLLMENT || fee < MIN_FEE) {
+            return false;
+        }
+        return true;
+    }
+
+
+    public boolean isCapacityFull(int enrolledStudentCount) {
+        if (priceType.isFree()) {
+            return false;
+        }
+        return maxEnrollment <= enrolledStudentCount;
+    }
+
+    public boolean isPaymentCorrect(int payment) {
+        if (priceType == PriceType.FREE) {
```

Comment:
```suggestion
        if (priceType.isFree()) {
```

ㅜㅜ

저는 이런경우 의미를 드러내는 메시지를 표현하는 방식이 좋은 것같아요.
```
private isTypeFree() {
   return priceType.isFree();
}
```

`무료강의야?`라는 메시지가 사실 가장 중요한거겠죠.
그럼, 구현세부사항은 더이상 외부에서 신경쓰지 않을수 있게 되겠죠 🤔 

> `의도를 드러내는 이름`,  `메시지 보내기`